internal/controllers/user_devices_controller_v2.go

```diff
@@ -0,0 +1,127 @@
+package controllers
+
+import (
+	"database/sql"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/DIMO-Network/shared"
+	"github.com/DIMO-Network/shared/db"
+	"github.com/ericlagergren/decimal"
+	"github.com/gofiber/fiber/v2"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/tidwall/gjson"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+	"github.com/volatiletech/sqlboiler/v4/types"
+
+	"github.com/DIMO-Network/devices-api/internal/config"
+	"github.com/DIMO-Network/devices-api/internal/controllers/helpers"
+	"github.com/DIMO-Network/devices-api/internal/services"
+	"github.com/DIMO-Network/devices-api/models"
+)
+
+type UserDevicesControllerV2 struct {
+	Settings     *config.Settings
+	DBS          func() *db.ReaderWriter
+	log          *zerolog.Logger
+	DeviceDefSvc services.DeviceDefinitionService
+}
+
+func NewUserDevicesControllerV2(settings *config.Settings, dbs func() *db.ReaderWriter, logger *zerolog.Logger, deviceDefSvc services.DeviceDefinitionService) UserDevicesControllerV2 {
+	return UserDevicesControllerV2{
+		Settings:     settings,
+		DBS:          dbs,
+		log:          logger,
+		DeviceDefSvc: deviceDefSvc,
+	}
+}
+
+// GetRange godoc
+// @Description gets the estimated range for a particular user device
+// @Tags        user-devices
+// @Produce     json
+// @Success     200 {object} controllers.DeviceRange
+// @Security    BearerAuth
+// @Param       tokenId path int true "tokenId"
+// @Router      /v2/vehicles/{tokenId}/analytics/range [get]
+func (udc *UserDevicesControllerV2) GetRange(c *fiber.Ctx) error {
```

Comment:
It looks like most of this is duplicate code from the v1/GetRange. If this is true, can you move it to a single function that both endpoints call?